### PR TITLE
EN-49813: query hint /*+ hints */

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

**Review by rjmac — CHANGES_REQUESTED** [review]
There's no reason to make these hints like comments.  Choose some other thing, putting semantic data in comments is a curse caused by needing to smuggle information past some other thing that doesn't understand it, and soql isn't in that position.

**Review by rjmac — CHANGES_REQUESTED** [review]
Also I see no validation or interpretation of these hints.  They need to be parsed here, not just blindly passed through.

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-ThisBuild / version := "3.15.0"
+version in ThisBuild := "4.0.0"
```
Comment (saraid) [inline version.sbt:1]: Did we downgrade sbt or something?

Comment (rjmac) [reply version.sbt:1]: Huh, I didn't notice that.  That shouldn't have changed back.  :eyes:

Comment (rjmac) [reply version.sbt:1]: Actually it doesn't look like this project uses sbt-release at all, so I'm not sure what changed that.

**Pull request closed without merging**